src/compile/concat.ts

```diff
@@ -77,6 +92,16 @@ export class ConcatModel extends Model {
 
     for (const child of this.children) {
       child.parseLegend();
+
+      keys(child.component.legends).forEach((channel: ScaleChannel) => {
+        if (this.resolve[channel].legend === 'shared') {
+          // just use the first legend definition for each channel
```

Comment:
Add `// TODO: merge properties if they are not conflicting`

(This is still wrong, but the fix is beyond the scope of this PR.)